### PR TITLE
Migrate the rest of the source code to PHP 8 syntax

### LINKED ISSUE #9772: Migrate to PHP 8.1
This is something that I think should go in milestone 3.0.0

A big difference between branches 2 and 3 is the `php` requirement:

```diff
- "php": "^7.1 || ^8.0",
+ "php": "^8.1",
```

Now that support for PHP 7 has been dropped, we can use features that are specific to PHP 8.

@derrabus has already done a lot about that. When this is over, we should be able to get rid of lines mentioning native type declarations in `phpcs.xml.dist`. It can also be used to figure out what is left to do.

We should rely on constructor property promotion when possible.

phpdoc that is found to be wrong on the way should be fixed on the next patch branch.
phpdoc that is found to be imprecise on the way should be improved on the next minor branch. 

Here is an example for `AbstractQuery` and related classes:
- patch: https://github.com/doctrine/orm/pull/9774
- minor: https://github.com/doctrine/orm/pull/9777
- major: https://github.com/doctrine/orm/pull/9776

### PR DESCRIPTION
**Pull request by greg0ire**
Closes #9772 (yes, finally!)

None of the remaining changes look legit.

### CODE REVIEW

File: lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php [hunk 1]
```diff
@@ -41,14 +41,14 @@ class XmlDriver extends FileDriver
...
-    private $isXsdValidationEnabled;
-
     /**
      * {@inheritDoc}
      */
-    public function __construct($locator, $fileExtension = self::DEFAULT_FILE_EXTENSION, bool $isXsdValidationEnabled = false)
-    {
+    public function __construct(
+        $locator,
+        $fileExtension = self::DEFAULT_FILE_EXTENSION,
```
Comment (derrabus) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php]: Types? 🤔

File: lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php [hunk 2]
```diff
@@ -41,14 +41,14 @@ class XmlDriver extends FileDriver
 {
     public const DEFAULT_FILE_EXTENSION = '.dcm.xml';
```
Comment (derrabus) [inline lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php:43]: `final`?

Comment (greg0ire) [reply lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php:43]: `SimplifiedXmlDriver` overrides it.

Comment (derrabus) [reply lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php:43]: 😭

**Review by derrabus — APPROVED** [review]

**Pull request merged**